test/sampling_correlation_matrices_test.cpp

```diff
@@ -64,6 +79,35 @@ void check_output(PointList &randPoints, int num_points, int n){
     CHECK(score.maxCoeff() < 1.1);
 }
 
+template<typename NT, typename VT, typename MT>
+void check_output_MT(std::list<MT> &randCorMatrices, int num_points, int n){
+	int d = n*(n-1)/2, count = 0;
+	MT A;
+	Eigen::LDLT<MT> A_ldlt;
+	for(auto& mat : randCorMatrices){
+    	    auto coeffs = getCoefficientsFromMatrix<NT, MT>(mat);
+    	    A = rebuildMatrix<NT, MT>(coeffs, n);
```

Comment:
A should be equal to mat here, isn't it?
I don't see any reason for `getCoefficientsFromMatrix()` and `rebuildMatrix()` since you want to check if `mat` is positive definite.

Reply:
Yes, I'm sorry that was a mistake i'll delete the two lines